src/lib/modules/ens/index.js

```diff
@@ -217,15 +217,19 @@ class ENS {
   }
 
   registerConfigDomains(config, cb) {
-    const self = this;
     const secureSend = embarkJsUtils.secureSend;
 
-    self.events.request("blockchain:defaultAccount:get", (defaultAccount) => {
-      async.each(Object.keys(self.registration.subdomains), (subDomainName, eachCb) => {
-        const address = self.registration.subdomains[subDomainName];
-        const reverseNode = utils.soliditySha3(address.toLowerCase().substr(2) + reverseAddrSuffix);
-        ENSFunctions.registerSubDomain(self.ensContract, self.registrarContract, self.resolverContract, defaultAccount,
-          subDomainName, self.registration.rootDomain, reverseNode, address, self.logger, secureSend, eachCb);
+    this.events.request("blockchain:defaultAccount:get", (defaultAccount) => {
+      async.each(Object.keys(this.registration.subdomains), (subDomainName, eachCb) => {
+        const address = this.registration.subdomains[subDomainName];
+        this.ensResolve(`${subDomainName}.${this.registration.rootDomain}`, (error, currentAddress) => {
+          if (error || currentAddress.toLowerCase() === address.toLowerCase()) {
```

Comment:
I would do another if for the error, above this one, and log the error if it happens. Because right now, if there is an error, the subdomain won't be registered and the user won't know it.

Reply:
Updated